async_processor/types.py

```diff
@@ -58,13 +73,25 @@ def get_body(self) -> Any:
     def should_stream_response(self) -> bool:
         return self.stream_response
 
+    def get_request_headers(self) -> Optional[Dict[str, Union[str, List[str]]]]:
+        return self.request_headers
+
+    def get_request_url(self) -> Optional[str]:
+        return self.request_url
+
+    def get_request_method(self) -> Optional[str]:
+        return self.request_method
+
 
 # We cannot maintain two different types and should remove `InputMessage`
 # after sometime
 class InputMessageV2(InputMessageInterface):
     tfy_request_id: Optional[constr(regex=r"^[a-zA-Z0-9\-]{1,36}$")] = None
     tfy_published_at_epoch_ns: Optional[int] = None
     tfy_stream_response: bool = False
+    tfy_request_headers: Optional[Dict[str, Union[str, List[str]]]] = None
```

Comment:
Header values can be list? 🤔

Reply:
One header can have multiple values. I will probably change the type later.